llvm/lib/Transforms/Vectorize/VPlanRecipes.cpp

```diff
@@ -107,6 +108,7 @@ bool VPRecipeBase::mayReadFromMemory() const {
     return cast<VPInstruction>(this)->opcodeMayReadOrWriteFromMemory();
   case VPWidenLoadEVLSC:
   case VPWidenLoadSC:
+  case VPWidenFFLoadSC:
```

Comment:
Does this need to get added to `VPRecipeBase::mayHaveSideEffects()` as well?

Reply:
I lean toward treating VPWidenFFLoad as side‑effecting since it returns vector length, similar to VPInstruction::ExplicitVectorLength.